src/ERC6538Registry.sol

```diff
@@ -9,13 +9,47 @@ contract ERC6538Registry is IERC6538Registry {
   /// @notice Maps a registrant's identifier to the scheme ID to the stealth meta-address.
   /// @dev `registrant` may be a standard 160-bit address or any other identifier.
   /// @dev `schemeId` is an integer identifier for the stealth address scheme.
-  mapping(bytes registrant => mapping(uint256 schemeId => bytes)) public stealthMetaAddressOf;
+  mapping(address registrant => mapping(uint256 schemeId => bytes stealthMetaAddress)) public
+    stealthMetaAddressOf;
+  mapping(address user => uint256 nonce) public nonceOf;
+
+  bytes32 private constant TYPE_HASH =
+    keccak256("EIP712Domain(string name,string version,uint256 chainId,address registryContract)");
+
+  // Cache the domain separator as an immutable value, but also store the chain id that it
+  // corresponds to, in order to
+  // invalidate the cached domain separator if the chain id changes.
+  bytes32 private immutable _cachedDomainSeparator;
+  uint256 private immutable _cachedChainId;
+  address private immutable _cachedThis;
+
+  bytes32 private immutable _hashedName;
+  bytes32 private immutable _hashedVersion;
+
+  string private _name;
+  string private _version;
+
+  enum RecoverError {
+    NoError,
+    InvalidSignature,
+    InvalidSignatureLength,
+    InvalidSignatureS
+  }
+
+  constructor() {
+    _name = "ERC6538Registry";
+    _version = "1";
+    _hashedName = keccak256(bytes("ERC6538Registry"));
+    _hashedVersion = keccak256(bytes("1"));
```

Comment:
```suggestion
    _hashedName = keccak256(bytes(_name));
    _hashedVersion = keccak256(bytes(_version));
```

Reply:
Updated in the same commit as above https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/4460b6edabe9487c5157a75f918ee7c116919931